src/ArborX_LinearBVH.hpp

```diff
@@ -131,15 +132,22 @@ class BoundingVolumeHierarchy
 public:
   using legacy_tree = void;
 
+  using bounding_volume_type = typename base_type::bounding_volume_type;
+
   BoundingVolumeHierarchy() = default; // build an empty tree
 
   template <typename ExecutionSpace, typename Primitives,
             typename SpaceFillingCurve = Experimental::Morton64>
   BoundingVolumeHierarchy(ExecutionSpace const &space,
                           Primitives const &primitives,
                           SpaceFillingCurve const &curve = SpaceFillingCurve())
-      : base_type(space, primitives, curve)
-  {}
+      : base_type(space,
+                  Details::LegacyValues{primitives, bounding_volume_type{}},
+                  Details::DefaultIndexableGetter(), curve)
+  {
+    // We want this before calling the base_type constructor
+    Details::check_valid_access_traits(PrimitivesTag{}, primitives);
```

Comment:
@dalg24 Not sure how to do this. The idea is that I'd like to check legacy things using AccessTraits, and then check things inside the new API using RangeTraits (not part of this PR). The problem is that that base class would be called before the access traits validation, so the user will see incomprehensible error.